daft/dataframe/dataframe.py

```diff
@@ -1060,37 +1081,35 @@ def agg_concat(self, *cols: ColumnInputType) -> "DataFrame":
         Returns:
             DataFrame: Globally aggregated list. Should be a single row.
         """
-        if len(cols) == 0:
-            warnings.warn(
-                "No columns specified; performing agg_concat on all columns. Specify columns using df.agg_concat('col1', 'col2', ...)."
-            )
-            cols = tuple(self.columns)
-        return self._agg([(c, "concat") for c in cols])
+        return self._apply_agg_fn(Expression.agg_concat, cols)
 
     @DataframePublicAPI
-    def agg(self, to_agg: List[Tuple[ColumnInputType, str]]) -> "DataFrame":
+    def agg(self, *to_agg: ColumnInputOrListType) -> "DataFrame":
```

Comment:
What would happen if a user supplied multiple expression lists, e.g. `df.agg([col("a").sum()], [col("b").mean()])`? By my reading, it would fail in the builder [here](https://github.com/Eventual-Inc/Daft/blob/0fc57d73eaaf91217e94ba63b2ab717d6f654a25/daft/logical/builder.py#L142), since the input would be `List[List[Expression]]`.

We could disallow this by eagerly checking and raising an error if any of `to_agg` are lists when `len(to_agg) > 1`; this would still be backwards compatible, since `df.agg()` previously only took a single argument.